extension/source/background/AggregatorController.ts

```diff
@@ -158,4 +175,55 @@ export default class AggregatorController {
       this.preferredAggregators[providerId] = preferredAggregator;
     },
   });
+
+  async addFeePaymentAction(
+    nonce: string,
+    actions: Array<ActionData>,
+    wallet: BlsWalletWrapper,
+    aggregator: Aggregator,
+  ) {
+    const network = await this.networkController.network.read();
+    const netCfg = getNetworkConfig(network, this.multiNetworkConfig);
+    const ethersProvider = await this.ethersProvider.read();
+
+    const aggregatorUtilitiesAddress = netCfg.addresses.utilities;
+    const aggregatorUtilitiesContract = AggregatorUtilitiesFactory.connect(
+      aggregatorUtilitiesAddress,
+      ethersProvider,
+    );
+
+    const actionsWithFeePaymentAction = [
+      ...actions,
+      {
+        ethValue: 1,
```

Comment:
Possibly a dumb question

Why is this action needed? And why do we set ethValue to 1?

Reply:
Oh nvm.  Is this added here just to get an accurate fee estimate?

Reply:
> why do we set ethValue to 1?

Not a dumb question at all, I had the exact same question before. I had a discussion with @voltrevo about this and it's possible to provide a value of 0 wei, or 1 wei for the `ethValue` when estimating fees. If you follow this thread from [this comment onwards](https://github.com/web3well/bls-wallet/pull/535#discussion_r1131800261), that should provide some useful context.

My understanding is that you need to add the payment action to the estimate fee bundle in order to prevent the [actual transaction costing more than the estimate](https://github.com/web3well/bls-wallet/blob/55ad789c3a86831d4bbadff92a402ae1d017f96c/aggregator/README.md?plain=1#L107-L110). Then it is a choice whether the `ethValue` is 0 or 1. The discussion I've linked above contains some of the tradeoffs for each option.